lib/beaker/dsl/helpers/host_helpers.rb

```diff
@@ -582,6 +582,37 @@ def echo_on hosts, val
             end
           end
         end
+
+        # Write the specified set of hosts to the given YAML file.
+        # If the file exists, then its 'HOSTS' key will be overwritten.
+        # Otherwise, the file will be created and will contain only the
+        # 'HOSTS' key.
+        #
+        # Note that this helper is especially useful with the beaker
+        # subcommand workflow because it lets you preserve hosts across
+        # subcommands. For example, you can use it to ensure that any
+        # type defaults set in the pre-suite carry over to the tests.
+        # To use this capability, invoke the helper as
+        #     write_hosts(hosts, Subcommands::SubcommandUtil::SUBCOMMAND_OPTIONS)
+        # in your tests.
+        #
+        # @param [Host, Array<Host>, String, Symbol] hosts    One or more hosts to act upon,
+        #                            or a role (String or Symbol) that identifies one or more hosts.
+        #                            This should almost always be the global 'hosts' variable.
+        # @param [String] file The file to write to. This should typically be an existent options file
+        #                 such as Subcommands::SubcommandUtil::SUBCOMMAND_OPTIONS
+        def write_hosts(hosts, file)
+          hosts_hash = {}
```

Comment:
Might be better to use the hosts_hash from the current `HOSTS` entry in the file, if it has it and then merge the new host hash into it? This way, if `hosts` is a subset of the global `hosts`, any non-affected hosts remain as-is. I'll defer to reviewers on this.

Reply:
Updated the PR to go ahead and do this since it cleans up https://github.com/puppetlabs/beaker-puppet/pull/77